### ISSUE TITLE
RuntimeError: ProcessGroupNCCL is only supported with GPUs, no GPUs found!

### ISSUE THREAD

**Issue by roperi** [bghira/SimpleTuner#237]
Running on a 1x 4090 

```
$ bash train_sdxl.sh                                                                                                                                                                

Deleting Jupyter notebook folders in 5 seconds if you do not cancel out.                                                                                                                                                         
These folders are generally useless, and will cause problems if they remain.                                                                                                                                                     
Use 'export PROTECT_JUPYTER_FOLDERS=1' to prevent this behaviour, before starting the script.                                                                                                                                    
Alternatively, place this value in your env file.                                                                                                                                                                                
.....                                                                                                                                                                                                                            
YOUR TIME HAS COME.                                                                                                                                                                                                              
find: ‘/notebooks/datasets/input/’: No such file or directory                                                                                                                                                                    
/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/torch/cuda/__init__.py:138: UserWarning: CUDA initialization: Unexpected error from cudaGetDeviceCount(). Did you run some cuda functions before calling NumCudaDevices
() that might have already set an error? Error 804: forward compatibility was attempted on non supported HW (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)                                                         
  return torch._C._cuda_getDeviceCount() > 0                                                                                                                                                                                     
/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/bitsandbytes/cextension.py:34: UserWarning: The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication, and GPU qua
ntization are unavailable.                                                                                                                                                                                                       
  warn("The installed version of bitsandbytes was compiled without GPU support. "                                                                                                                                                
/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so: undefined symbol: cadam32bit_grad_fp32                                                                                            
[2023-12-13 22:12:34,193] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)                                                                                                          
/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/torch/cuda/__init__.py:138: UserWarning: CUDA initialization: Unexpected error from cudaGetDeviceCount(). Did you run some cuda functions before calling NumCudaDevices
() that might have already set an error? Error 804: forward compatibility was attempted on non supported HW (Triggered internally at ../c10/cuda/CUDAFunctions.cpp:108.)                                                         
  return torch._C._cuda_getDeviceCount() > 0                                                                                                                                                                                     
/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/bitsandbytes/cextension.py:34: UserWarning: The installed version of bitsandbytes was compiled without GPU support. 8-bit optimizers, 8-bit multiplication, and GPU qua
ntization are unavailable.                                                                                                                                                                                                       
  warn("The installed version of bitsandbytes was compiled without GPU support. "                                                                                                                                                
/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/bitsandbytes/libbitsandbytes_cpu.so: undefined symbol: cadam32bit_grad_fp32                                                                                            
[2023-12-13 22:12:36,157] [INFO] [real_accelerator.py:158:get_accelerator] Setting ds_accelerator to cuda (auto detect)                                                                                                          
2023-12-13 22:12:36,719 [INFO] (ArgsParser) VAE Cache location: /notebooks/datasets/models/cache_vae                                                                                                                             
2023-12-13 22:12:36,719 [INFO] (ArgsParser) Text Cache location: cache                                                                                                                                                           
[2023-12-13 22:12:36,721] [INFO] [comm.py:637:init_distributed] cdb=None                                                                                                                                                         
[2023-12-13 22:12:36,721] [INFO] [comm.py:668:init_distributed] Initializing TorchBackend in DeepSpeed with backend nccl                                                                                                         
Traceback (most recent call last):                                                                                                                                                                                       [5/1974]
  File "/workspace/SimpleTuner/train_sdxl.py", line 1426, in <module>                                                                                                                                                            
    main()                                                                                                                                                                                                                       
  File "/workspace/SimpleTuner/train_sdxl.py", line 164, in main                                                                                                                                                                 
    accelerator = Accelerator(                                                                                                                                                                                                   
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/accelerate/accelerator.py", line 358, in __init__                                                                                                              
    self.state = AcceleratorState(
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/accelerate/state.py", line 720, in __init__
    PartialState(cpu, **kwargs)
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/accelerate/state.py", line 170, in __init__
    dist.init_distributed(dist_backend=self.backend, auto_mpi_discovery=False, **kwargs)
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/deepspeed/comm/comm.py", line 670, in init_distributed
    cdb = TorchBackend(dist_backend, timeout, init_method, rank, world_size)
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/deepspeed/comm/torch.py", line 116, in __init__
    self.init_process_group(backend, timeout, init_method, rank, world_size)
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/deepspeed/comm/torch.py", line 142, in init_process_group
    torch.distributed.init_process_group(backend,
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/torch/distributed/c10d_logger.py", line 74, in wrapper
    func_return = func(*args, **kwargs)
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py", line 1148, in init_process_group
    default_pg, _ = _new_process_group_helper(
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/torch/distributed/distributed_c10d.py", line 1279, in _new_process_group_helper
    backend_class = ProcessGroupNCCL(backend_prefix_store, group_rank, group_size, pg_options)
RuntimeError: ProcessGroupNCCL is only supported with GPUs, no GPUs found!
[2023-12-13 22:12:39,483] torch.distributed.elastic.multiprocessing.api: [ERROR] failed (exitcode: 1) local_rank: 0 (pid: 5484) of binary: /workspace/SimpleTuner/.venv/bin/python
Traceback (most recent call last):
  File "/workspace/SimpleTuner/.venv/bin/accelerate", line 8, in <module>
    sys.exit(main())
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/accelerate/commands/accelerate_cli.py", line 45, in main
    args.func(args)
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/accelerate/commands/launch.py", line 964, in launch_command
    deepspeed_launcher(args)
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/accelerate/commands/launch.py", line 687, in deepspeed_launcher
    distrib_run.run(args)
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/torch/distributed/run.py", line 797, in run
    elastic_launch(
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 134, in __call__
    return launch_agent(self._config, self._entrypoint, list(args))
  File "/workspace/SimpleTuner/.venv/lib/python3.10/site-packages/torch/distributed/launcher/api.py", line 264, in launch_agent
    raise ChildFailedError(
torch.distributed.elastic.multiprocessing.errors.ChildFailedError: 
============================================================
train_sdxl.py FAILED
------------------------------------------------------------
Failures:
  <NO_OTHER_FAILURES>
------------------------------------------------------------
Root Cause (first observed failure):
[0]:
  time      : 2023-12-13_22:12:39
  host      : ff06483c1709
  rank      : 0 (local_rank: 0)
  exitcode  : 1 (pid: 5484)
  error_file: <N/A>
  traceback : To enable traceback see: https://pytorch.org/docs/stable/elastic/errors.html
============================================================

```

Deepseed config:

 cat /root/.cache/huggingface/accelerate/default_config.yaml 
```
compute_environment: LOCAL_MACHINE
deepspeed_config:
  gradient_accumulation_steps: 4
  zero3_init_flag: false
  zero_stage: 1
distributed_type: DEEPSPEED
downcast_bf16: 'no'
machine_rank: 0
main_training_function: main
mixed_precision: bf16
num_machines: 1
num_processes: 1
rdzv_backend: static
same_network: true
tpu_env: []
tpu_use_cluster: false
tpu_use_sudo: false
use_cpu: false
```

**Comment by bghira** [comment]
env issues are out of scope for this project tbh

**Comment by bghira** [comment]
nvidia stuff is definitely a pain in the butt. there's an ongoing issue report for eternity in the bitsandbytes project where it just doesn't work and it's the same reason, either some kind of mismatched OS library or something. one thing that stands out here is you're using an 11.7 CUDA image and then installing 11.8 libraries

**Comment by bghira** [comment]
if it works before, i think maybe some kind of OS update that's getting pulled in might be messing with things. did it do that more than once on different hw? it _could_ be a hw issue/issue with the host OS config

**Comment by bghira** [comment]
thanks for working through that, this isn't a workflow i'm capable of testing/fixing issues for.

**Comment by bghira** [comment]
i would suggest not installing that, then

**Comment by bghira** [comment]
i'm using pytorch 2.1 with cuda 12.3 locally on a 4090
